### PR TITLE
Custom mailer for dynamic subjects

### PR DESCRIPTION
**Pull request by sansbrina**
## Why is this PR needed?
Closes https://github.com/CZagrobelny/new_sanctuary_asylum/issues/242 by adding a custom devise mailer that adds the community name to the subject line of the invitation instructions email.

## Solution
I followed [the devise_invitable docs](https://github.com/scambra/devise_invitable/wiki/Customizing-for-different-Invite-use-cases-(emails-etc.)) for this one.

### Link to associated issue: 
https://github.com/CZagrobelny/new_sanctuary_asylum/issues/242

### CODE REVIEW

File: app/mailers/invite_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class InviteMailer < Devise::Mailer
+
+  protected
```
Comment (sansbrina) [inline app/mailers/invite_mailer.rb]: This is protected, so I figured there was no need to add a spec for it (let me know if that's a wrong assumption at all.)

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
   config.mailer_sender = ENV['FROM_ADDRESS']
 
   # Configure the class responsible to send e-mails.
-  config.mailer = 'Devise::Mailer'
+  config.mailer = 'InviteMailer'
 
   # Configure the parent class responsible to send e-mails.
   # config.parent_mailer = 'ActionMailer::Base'
```
Comment (sansbrina) [inline config/initializers/devise.rb]: Note: I _think_ the parent class becomes Devise::Mailer in this case.

File: app/mailers/invite_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  def subject_for(key)
+    if key.to_s == 'invitation_instructions'
+      "New Sanctuary Coalition (#{ @resource.community.name.upcase }) volunteer account invitation"
+    else
+      return super
+    end
```
Comment (sansbrina) [inline app/mailers/invite_mailer.rb]: Happy to refactor the above block if there's a different style you'd like to go for here, was going off the docs for simplicity.

**Comment by CZagrobelny** [comment]
@sansbrina this is so exciting, I am looking forward to reviewing!  I have been a little slow, since I've been at a conference this week, will definitely review this weekend or earlier!

**Pull request merged**